deploy/00_resolve_keep_token.ts

```diff
@@ -0,0 +1,48 @@
+import { HardhatRuntimeEnvironment, HardhatNetworkConfig } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+import { BigNumber } from "ethers"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { getNamedAccounts, deployments, helpers } = hre
+  const { log } = deployments
+  const { deployer } = await getNamedAccounts()
+  const { execute, read } = deployments
+
+  const KeepToken = await deployments.getOrNull("KeepToken")
+
+  if (KeepToken && helpers.address.isValid(KeepToken.address)) {
+    log(`using external KeepToken at ${KeepToken.address}`)
+  } else if (
+    hre.network.name !== "hardhat" ||
+    (hre.network.config as HardhatNetworkConfig).forking.enabled
+  ) {
+    throw new Error("deployed KeepToken contract not found")
+  } else {
+    // For deployments on hardhat network we don't have the KeepToken deployed,
+    // so we're deploying a stub contact and minting the KEEP in the amount
+    // close to the KEEP supply on the production environment (~1B KEEP).
+    log(`deploying KeepToken stub`)
+
+    await deployments.deploy("KeepToken", {
+      contract: "TestToken",
+      from: deployer,
+      log: true,
+    })
+
+    await execute(
+      "KeepToken",
+      { from: deployer },
+      "mint",
+      deployer,
+      BigNumber.from(10).pow(27)
```

Comment:
1. Let's extract the value to a constant.
2. We could use a helper function to handle the precision: 
  ```js
import { to1e18 } from "../test/helpers/contract-test-helpers"

const KEEP_SUPPLY = to1e18("1000000000") // 1B KEEP
```

Reply:
Done, see 3c36d41.